src/views/AddHr.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+
+interface FormValues {
+  email: string;
+  name: string;
+  company: string;
+}
+
+const schema = yup.object().shape({
+  email: yup.string().email('Email musi mieć odpowiedni format').required('Email jest wymagany'),
+  name: yup.string().required('Imię jest wymagane').min(8).max(150),
+  company: yup.string().required('Nazwa firmy jest wymagana').min(8).max(150),
+});
+
+export const AddHr: React.FC = () => {
+  const defaultValues = {
+    email: '',
+    name: '',
+    company: '',
+  };
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormValues>({ resolver: yupResolver(schema), defaultValues });
+
+  const onSubmit = (data: FormValues) => {
+    console.log(data);
+  };
+
+  return (
```

Comment:
Też bym to podzielił tutaj na mniejsze komponenty, ale tbh nie wiem, czy mamy na to czas.